File: api/mathew/entities/Reimbursement.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace mathew.entities;

public class Reimbursement
{
    public int Id { get; set; }

    [Required]
    [Column(TypeName = "decimal(18,2)")]
    public decimal Amount { get; set; }

    [Required]
    public int ExpenseId { get; set; }
    public Expense Expense { get; set; }

    [Required]
    public string UserName { get; set; }

    [Required]
    public int Percentage { get; set; }

    [Required] public bool Pending { get; set; }
}

File: ng/src/app/models/reimbursement.ts
import { Expense } from '@/models/expense';

export class Reimbursement {

    constructor() {
        this.id = 0;
        this.amount = 0;
        this.expenseId = 0;
        this.userName = '';
        this.familyId = 0;
        this.description = '';
        //this.percentage = 0;
        // this.pending = true;
    }

    id: number;
    amount: number;
    expenseId: number;
    userName: string;
    percentage?: number;
    pending?: boolean;
    expense?: Expense;
    familyId: number;
    reimbursementDate?: Date;
    description: string;
}
